File: system-test/java/src/main/java/com/optivem/shop/dsl/port/given/steps/GivenOrder.java
package com.optivem.shop.dsl.port.given.steps;

import com.optivem.shop.dsl.port.given.steps.base.GivenStep;
import com.optivem.shop.dsl.driver.port.shop.dtos.OrderStatus;

public interface GivenOrder extends GivenStep {
    GivenOrder withOrderNumber(String orderNumber);

    GivenOrder withSku(String sku);

    GivenOrder withQuantity(String quantity);

    GivenOrder withQuantity(int quantity);

    GivenOrder withStatus(String status);

    GivenOrder withStatus(OrderStatus status);
}

File: system-test/typescript/src/testkit/dsl/port/given/steps/given-order.ts
import type { GivenStage } from '../given-stage.js';
import type { WhenStage } from '../../when/when-stage.js';
import type { ThenStage } from '../../then/then-stage.js';

export interface GivenOrder {
  withOrderNumber(orderNumber: string): this;
  withSku(sku: string): this;
  withQuantity(quantity: string | number): this;
  withCountry(country: string): this;
  withCouponCode(couponCode: string | null): this;
  withStatus(status: string): this;
  and(): GivenStage;
  when(): WhenStage;
  then(): ThenStage;
}
